src/connection.js

```diff
@@ -405,6 +422,62 @@ export default class Connection extends EventEmitter {
     return this.messageIo.on('secure', this.emit.bind(this, 'secure'));
   }
 
+  resolveIPaddresses(host, callback) {
+    let ipv4Addresses = undefined;
+    let ipv6Addresses = undefined;
+    //resolve IPV4
+    Dns.resolve4(host, (err, addresses) => {
+      ipv4Addresses = addresses || new Array();
+      if (ipv6Addresses) {
+        callback(ipv4Addresses.concat(ipv6Addresses));
+      }
+    });
+    
+    //resolve IPV6
+    Dns.resolve6(host, (err, addresses) => {
+      ipv6Addresses = addresses || new Array();
+      if (ipv4Addresses) {
+        callback(ipv4Addresses.concat(ipv6Addresses));
+      }
+    });
+  }
+ 
+  multiConnectOnPort(addresses, port, callback) {
```

Comment:
The ```multiConnectOnPort ``` function only figures out which IP address resolved from dns can be connected, and invoke callback in ```connectOnPort```, which calls ```socket.connect``` using connection opts and establish the connection.

I could try to move the logic of 'establish connection using ```this.socket.connect```' from ```connectOnPort``` to ```multiConnectOnPort ```. Do you think it will make the implementation clearer? :smile: